### PR TITLE
chore: deduplicate AccessList and Withdrawals types

### PR DESCRIPTION
**Pull request by JackG-eth**
closes https://github.com/alloy-rs/alloy/issues/255

deduplicates AccessList+Withdrawals by moving them into eips crate and reusing them in rpc and consensus

### CODE REVIEW

File: crates/eips/src/withdrawal.rs
Comment (mattsse) [inline crates/eips/src/withdrawal.rs:1]: let's rename this eip4895.rs and documents, see also eip2930.rs

https://eips.ethereum.org/EIPS/eip-4895

this file also needs changes so serde is optional and we need ssz feature

**Review by mattsse — CHANGES_REQUESTED** [review]
great start,

a few more todos!

File: crates/eips/Cargo.toml [hunk 1]
```diff
@@ -14,9 +14,11 @@ exclude.workspace = true
 [dependencies]
 alloy-primitives = { workspace = true, features = ["rlp"], default-features = false }
 alloy-rlp = { workspace = true, features = ["derive"] }
+alloy-serde.workspace = true
 
 # serde
 serde = { workspace = true, default-features = false, optional = true }
+serde_json.workspace = true
```
[the remark this answers is not in the document]
Comment (JackG-eth) [reply crates/eips/Cargo.toml]: moved

File: crates/eips/src/withdrawal.rs
Comment (JackG-eth) [reply crates/eips/src/withdrawal.rs:1]: Added optionality of serde, is this as some packages need it and others don't pushing more performance?

Comment (JackG-eth) [reply crates/eips/src/withdrawal.rs:1]: Do you mean szz feature for eip2930? as it already exists for withdrawals.

File: crates/eips/src/eip4895.rs [hunk 2]
```diff
@@ -1,27 +1,17 @@
...
-    Default,
-    Hash,
-    RlpEncodable,
-    RlpDecodable,
-    Serialize,
-    Deserialize,
-)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Default, Hash, RlpEncodable, RlpDecodable)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "ssz", derive(ssz_derive::Encode, ssz_derive::Decode))]
```
Comment (mattsse) [inline crates/eips/src/eip4895.rs:15]: this feature currently does not exist in this crate, so we need to add this and the required deps, see rpc-types for reference
then we have to enable this in rpc-types if the ssz feature is enabled there

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
fyi @klkvr with this AccessList types are deduplicated

**Comment by onbjerg** [comment]
can you also remove https://github.com/alloy-rs/alloy/blob/e8c622c713938cd2ede29d6b38d2aaa009dd3ad5/crates/network/src/ethereum/builder.rs#L229 which was part of the motivation for this PR?

**Review by onbjerg — CHANGES_REQUESTED** [review]
(see above)

**Comment by mattsse** [comment]
done

**Review by onbjerg — APPROVED** [review]
lgtm

**Pull request merged**